sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -521,12 +525,37 @@ private static Exception instantiateUnexpectedException(UnexpectedExceptionInfor
 
                 switch (paramCount) {
                     case 3:
-                        return ctor.newInstance(httpRequest, responseStatusCode, responseHeaders);
+                        return Mono.just(ctor.newInstance(httpRequest, responseStatusCode, responseHeaders));
                     case 4:
-                        return ctor.newInstance(httpRequest, responseStatusCode, responseHeaders, bodyAsObject);
+                        return Mono.just(ctor.newInstance(httpRequest, responseStatusCode, responseHeaders,
+                            bodyAsObject));
                     case 5:
-                        return ctor.newInstance(httpRequest, responseStatusCode, responseHeaders, bodyAsObject,
-                            response.getDecodedHeaders().block());
+                        return response.getDecodedHeaders()
+                            .defaultIfEmpty(NullObject.INSTANCE)
+                            .map(headers -> {
```

Comment:
Should this be switched to a `flatMap` and follow a similar pattern as `handleRestResponseReturnType` which you changed in the PR as well.

Reply:
map works well in this case and make code less overhead.
I would prefer to keep using map here.